optimum/habana/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -571,3 +577,131 @@ def forward(
             attentions=transformer_outputs.attentions,
             cross_attentions=transformer_outputs.cross_attentions,
         )
+
+
+class GaudiGPT2DoubleHeadsModel(GPT2DoubleHeadsModel):
```

Comment:
please move this to https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/models/gpt2/modeling_gpt2.py and check how to overwrite functions in transformers with git blame to find PR.